### ISSUE TITLE
a experiment about meta-test

### ISSUE THREAD

**Issue by wjczf123** [microsoft/vert-papers#46]
Hi. I delete Line 384-385 and Line 447 of learner.py to avoid fine-tuning in the support set during the meta-test? Is this right? Thanks.

**Comment by iofu728** [comment]
Hi @wjczf123, yeah. If you remove the [#384-385](https://github.com/microsoft/vert-papers/blob/master/papers/DecomposedMetaNER/learner.py#L384-L385) and [#447](https://github.com/microsoft/vert-papers/blob/master/papers/DecomposedMetaNER/learner.py#L447) of learner.py,  the code'll skip fine-tuning on meta-test support set.

**Comment by wjczf123** [comment]
I understand the performance will drop, but it performs poorly.

**Comment by iofu728** [comment]
> Thanks for your reply. I ran it once under inter 5-way 1-shot setting and the results looked very bad.
> 
> 2022-09-20 22:57:35 INFO: - span_f1 = 0.7218073781712385 2022-09-20 22:57:35 INFO: - span_p = 0.7370060346505719 2022-09-20 22:57:35 INFO: - span_r = 0.7072229140722269 2022-09-20 22:57:35 INFO: - type_f1 = 0.156973848019738 2022-09-20 22:57:35 INFO: - type_p = 0.156973848069738 2022-09-20 22:57:35 INFO: - type_r = 0.156973848069738 2022-09-20 22:57:35 INFO: - 9.445,9.063,9.250,73.701,70.722,72.181,15.697,15.697,15.697,0.000,0.000,0.000

Sorry, I made a mistake earlier. You can't direct remove #447 in the type classification stage, which has some logit to generate the type embedding.
The solution should be keep [#447](https://github.com/microsoft/vert-papers/blob/master/papers/DecomposedMetaNER/learner.py#L447), and change [#165](https://github.com/microsoft/vert-papers/blob/master/papers/DecomposedMetaNER/learner.py#L165) to `self.model.eval()`. You may also need to remove [#191-192](https://github.com/microsoft/vert-papers/blob/master/papers/DecomposedMetaNER/learner.py#L191-L192)

**Comment by wjczf123** [comment]
Thanks. The new result seems to be correct.

2022-09-24 20:43:14 INFO: - ***** Eval results inter-test *****
2022-09-24 20:43:14 INFO: -   f1 = 0.6104350036041772
2022-09-24 20:43:14 INFO: -   f1_threshold = 0.6133144703132174
2022-09-24 20:43:14 INFO: -   loss = tensor(4.1757, device='cuda:0')
2022-09-24 20:43:14 INFO: -   precision = 0.6232885601193933
2022-09-24 20:43:14 INFO: -   precision_threshold = 0.6340790479672884
2022-09-24 20:43:14 INFO: -   recall = 0.5981008717310069
2022-09-24 20:43:14 INFO: -   recall_threshold = 0.5938667496886657
2022-09-24 20:43:14 INFO: -   span_f1 = 0.7218073781712385
2022-09-24 20:43:14 INFO: -   span_p = 0.7370060346505719
2022-09-24 20:43:14 INFO: -   span_r = 0.7072229140722269
2022-09-24 20:43:14 INFO: -   type_f1 = 0.8474159401741568
2022-09-24 20:43:14 INFO: -   type_p = 0.8474159402241568
2022-09-24 20:43:14 INFO: -   type_r = 0.8474159402241568
2022-09-24 20:43:14 INFO: - 62.329,59.810,61.044,73.701,70.722,72.181,84.742,84.742,84.742,63.408,59.387,61.331

**Comment by wjczf123** [comment]
I am very sorry to interrupt again. The 5-way 5-shot results under inter setting are worse than the 1-shot.

The results under 5-shot are:

2022-09-27 12:34:24 INFO: - ***** Eval results inter-test *****
2022-09-27 12:34:24 INFO: - f1 = 0.4713394738913809
2022-09-27 12:34:24 INFO: - f1_threshold = 0.4678844439614423
2022-09-27 12:34:24 INFO: - loss = tensor(5.6218, device='cuda:0')
2022-09-27 12:34:24 INFO: - precision = 0.6559220937309791
2022-09-27 12:34:24 INFO: - precision_threshold = 0.680110921612789
2022-09-27 12:34:24 INFO: - recall = 0.36782895994320486
2022-09-27 12:34:24 INFO: - recall_threshold = 0.3566065041094392
2022-09-27 12:34:24 INFO: - span_f1 = 0.5238582587165661
2022-09-27 12:34:24 INFO: - span_p = 0.7290079123554465
2022-09-27 12:34:24 INFO: - span_r = 0.4088141331949865
2022-09-27 12:34:24 INFO: - type_f1 = 0.9226715451538331
2022-09-27 12:34:24 INFO: - type_p = 0.9226715452038331
2022-09-27 12:34:24 INFO: - type_r = 0.9226715452038331
2022-09-27 12:34:24 INFO: - 65.592,36.783,47.134,72.901,40.881,52.386,92.267,92.267,92.267,68.011,35.661,46.788

The recall of span detection seems to low.

**Comment by wjczf123** [comment]
I am very sorry to interrupt again. Why is the performance in 5-shot worse than 1-shot after ablating fine-tuning in meta-test? 
For example, the F1 under inter 5way-5shot is about 54. However, the performance under 1-shot is 61.04. Have you ever observed this phenomenon before? It doesn't seem normal. Thanks.

**Comment by iofu728** [comment]
Hi @wjczf123, this may be reasonable, although we have not done the corresponding ablation experiments on 5shot. First of all the 5shot and 1shot datasets cannot be compared in parallel, they are both just a sampled subset of Few-NERD. Of course, according to our experimental results on inter 5-1 and inter 5-5, it seems that the 5shot results are better. Secondly, we found in our experiments that more fine-tuning steps are needed for inter 5-5 and inter 10-5 in the meta-test. Removing the fine-tune may have a greater impact on 5shot. Hope this helps.

**Comment by wjczf123** [comment]
Thanks. Hope you have a good day.
